src/main/java/com/zendesk/maxwell/replication/BinlogConnectorReplicator.java

```diff
@@ -492,10 +522,12 @@ public RowMap getRow() throws Exception {
 					QueryEventData qe = event.queryData();
 					String sql = qe.getSql();
 					if (BinlogConnectorEvent.BEGIN.equals(sql)) {
-						rowBuffer = getTransactionRows(event);
-						rowBuffer.setServerId(event.getEvent().getHeader().getServerId());
-						rowBuffer.setThreadId(qe.getThreadId());
-						rowBuffer.setSchemaId(getSchemaId());
+						try {
+							rowBuffer = getTransactionRows(event);
+							rowBuffer.setServerId(event.getEvent().getHeader().getServerId());
+							rowBuffer.setThreadId(qe.getThreadId());
+							rowBuffer.setSchemaId(getSchemaId());
+						} catch ( ClientReconnectedException e ) {}
```

Comment:
When this throws, presumably it's from `getTransactionRows()`? In which case `rowBuffer` remains whatever was in there last time around and we carry on processing, is that a problem? Should we `null` it in the catch block or something?

Reply:
if you look at line 481 or so, it's more or less guaranteed that `rowBuffer` is null or empty by the time we reach this part of the code (part of the goofy code structure that inverts pushing into polling).  I don't mind and explicit `rowBuffer = null` but it's not strictly necessary afaict.

Reply:
Ah right, `rowBuffer` never sticks around because we drain it at the start of each loop. I believe you, although if a later refactoring made it possible for the following lines (`rowBuffer.set*`) to throw, it would become a problem. I've opened #1191 for the sake of paranoia.